sources/web/datalab/static/datalab.js

```diff
@@ -277,6 +277,168 @@ function initializePage(dialog, saveFn) {
   });
 }
 
+// constants for minitoolbar operations
+const CELL_METADATA_COLLAPSED = 'hiddenCell';
+const CELL_METADATA_OUTPUT_COLLAPSED = 'collapsed';
+const COLLAPSE_BUTTON_CLASS = 'fa-eye';
+const UNCOLLAPSE_BUTTON_CLASS = 'fa-eye-slash';
+const COLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-compress';
+const UNCOLLAPSE_OUTPUT_BUTTON_CLASS = 'fa-expand';
+const CLEAR_BUTTON_CLASS = 'fa-minus-square-o';
+const RUN_BUTTON_CLASS = 'fa-play';
+const COLLAPSED_CELL_INNER_HTML = '-- Collapsed cell --';
+
+/**
+ * Collapse entire cell
+ */
+function toggleCollapseCell(cell, hide) {
+  if (!(CELL_METADATA_COLLAPSED in cell.metadata)) {
+    return;
+  }
+  let isHidden = typeof(hide) !== 'undefined' ? !hide : cell.metadata[CELL_METADATA_COLLAPSED];
+  let inputDiv = cell.cell_type === 'code' ? 'div.input' : 'div.inner_cell';
+  collapseSpan = cell.element.find('span.collapse-cell')[0];
+  if (isHidden) {
+    cell.element.find(inputDiv).show();
+    // for code cells, also show the output div
+    if (cell.cell_type == 'code')
+      cell.element.find('div.output_wrapper').show();
+    cell.element.find('div.cellPlaceholder').hide();
+    collapseSpan.classList.add(UNCOLLAPSE_BUTTON_CLASS);
+    collapseSpan.classList.remove(COLLAPSE_BUTTON_CLASS);
+  } else {
+    cell.element.find(inputDiv).hide();
+    // for code cells, also hide the output div
+    if (cell.cell_type == 'code')
+      cell.element.find('div.output_wrapper').hide();
+    cell.element.find('div.cellPlaceholder').show();
+    collapseSpan.classList.remove(UNCOLLAPSE_BUTTON_CLASS);
+    collapseSpan.classList.add(COLLAPSE_BUTTON_CLASS);
+  }
+  cell.metadata[CELL_METADATA_COLLAPSED] = !isHidden;
+}
+
+/**
+ * Collapse output of a cell
+ */
+function toggleCollapseCellOutput(cell, hide) {
+  if (!(CELL_METADATA_OUTPUT_COLLAPSED in cell.metadata)) {
+    return;
+  }
+  let isHidden = typeof(hide) !== 'undefined' ? !hide : cell.metadata[CELL_METADATA_OUTPUT_COLLAPSED];
```

Comment:
Same comment here about replacing usage of 'isHidden' with references to the 'hide' variable (while dealing with the fact that those are negated).